src/prometheus/deploy/prometheus-configmap.yaml.template

```diff
@@ -36,6 +36,13 @@ data:
     scrape_configs:
     - job_name: 'node_exporter'
       scrape_interval: {{ prom_info['scrape_interval']|default(30) }}s
+      {% if clusterinfo['api-server']['tls_access'] -%}
+      tls_config:
+        ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
```

Comment:
>/var/run/secrets/kubernetes.io/serviceaccount/ca.crt [](start = 17, length = 52)

maybe you should make the path to to keep the ca file and secretes configurable. Remember AKS is not the only k8s deployment you need to support.

Reply:
This change not only affect Prometheus and watchdog. It makes some assumption to the underlying host environment. 

I suggest you to break this into multiple PRs. 1. one makes the CA/secretes path configurable. 2. the other one allows Prometheus to specify the path to CA/Secretes. 3. the final one does the configuration for watchdog.

---
In reply to: [234093754](https://github.com/Microsoft/pai/pull/1710#discussion_r234093754) [](ancestors = 234093754)

Reply:
Will split it and resubmit

Reply:
@ydye how about define the path env and which i can get the path from the env.